pkg/prometheus/resource_selector.go

```diff
@@ -862,6 +871,33 @@ func (rs *ResourceSelector) validateKubernetesSDConfigs(ctx context.Context, sc
 			}
 		}
 
+		allowedSelectors := map[monitoringv1alpha1.Role][]string{
+			RolePod:           {string(RolePod)},
+			RoleService:       {string(RoleService)},
+			RoleEndpointSlice: {string(RolePod), string(RoleService), string(RoleEndpointSlice)},
+			RoleEndpoint:      {string(RolePod), string(RoleService), string(RoleEndpoint)},
+			RoleNode:          {string(RoleNode)},
+			RoleIngress:       {string(RoleIngress)},
+		}
+
+		for _, s := range config.Selectors {
+			if _, ok := allowedSelectors[config.Role]; !ok {
```

Comment:
we need strings.ToLower too?
```suggestion
			if _, ok := allowedSelectors[strings.ToLower(config.Role)]; !ok {
```

Reply:
@simonpasquier done